### PR TITLE
Comments and finished to split the UI from the Plugin

### PR DESCRIPTION
**Pull request by GDayne**

### CODE REVIEW

File: src/DrTests-CommentsToTests/CommentTestCase.class.st [hunk 1]
```diff
@@ -41,6 +46,17 @@ CommentTestCase class >> errorComment: stringComment class: aClass selector: aSy
 		yourself
 ]
 
+{ #category : #example }
+CommentTestCase class >> example [
+	"1 + 1 >>> 2"
+	^ CommentTestCase comment: '1 + 1 >>> 2' class: CommentTestCase class selector: #example
+]
+
+{ #category : #'as yet unclassified' }
```
Comment (juliendelplanque) [inline src/DrTests-CommentsToTests/CommentTestCase.class.st]: This need to be classified.
```suggestion
{ #category : #'accessing' }
```

File: src/DrTests-CommentsToTests/DTCommentToTest.class.st [hunk 2]
```diff
@@ -17,25 +21,15 @@ DTCommentToTest class >> weight [
...
-										action: [ "TODO: self reRunResultGroup: resultGroup" ] ] ] ] ].
-	^ results
+DTCommentToTest >> itemsToBeAnalysedFor: packagesSelected [
+	^ packagesSelected
+		flatCollect: [ :p | 
+			p definedClasses
+				select: [ :c | 
+					c methods
+						anySatisfy: [ :m | 
+							((m methodClass commentsAt: m selector)
```
Comment (juliendelplanque) [inline src/DrTests-CommentsToTests/DTCommentToTest.class.st:31]:
```suggestion
							((c commentsAt: m selector)
```

File: src/DrTests-CommentsToTests/DTCommentToTest.class.st [hunk 3]
```diff
@@ -17,25 +21,15 @@ DTCommentToTest class >> weight [
...
+								select: [ :com | com includesSubstring: '>>>' ]) isNotEmpty ] ] ]
```
Comment (juliendelplanque) [inline src/DrTests-CommentsToTests/DTCommentToTest.class.st]:
```suggestion
								anySatisfy: [ :com | com includesSubstring: '>>>' ]) ] ] ]
```

File: src/DrTests-Tests/CommentsToTestsTest.class.st [hunk 4]
```diff
@@ -1,3 +1,6 @@
+"
+A DrTestsUITest is a test class for testing the behavior of DrTests-CommentsToTestsTest
```
Comment (juliendelplanque) [inline src/DrTests-Tests/CommentsToTestsTest.class.st:2]:
```suggestion
A DrTestsUITest is a test class for testing the behavior of DrTests-CommentsToTestsTest.
```

File: src/DrTests-Tests/CommentsToTestsUITest.class.st [hunk 5]
```diff
@@ -1,3 +1,6 @@
+"
+A DrTestsUITest is a test class for testing the behavior of DrTests-CommentsToTestsTest  UI
```
Comment (juliendelplanque) [inline src/DrTests-Tests/CommentsToTestsUITest.class.st]:
```suggestion
A DrTestsUITest is a test class for testing the behavior of DrTests-CommentsToTestsTest UI.
```

File: src/DrTests-Tests/DTMockPlugin.class.st [hunk 6]
```diff
@@ -1,3 +1,6 @@
+"
+A Mock object used in test
```
Comment (juliendelplanque) [inline src/DrTests-Tests/DTMockPlugin.class.st:2]:
```suggestion
A Mock object used in test.
```

File: src/DrTests-CommentsToTests/DTCommentToTest.class.st [hunk 7]
```diff
@@ -62,19 +56,33 @@ DTCommentToTest >> resultButtonLabel [
...
 { #category : #api }
 DTCommentToTest >> runSuite: aTestSuite withResult: aResult [
 	aTestSuite
 		subscribe: TestAnnouncement
 		do: [ :testAnnouncement | 
-			self ui
-				updateStatus: 'Running test {1}.'
-				format: {testAnnouncement test asString} ].
+			self announcer
+				announce: (DTStatusUpdate message: ('Running test {1}.' format: {testAnnouncement test asString})) ].
```
Comment (guillep) [inline src/DrTests-CommentsToTests/DTCommentToTest.class.st:85]: Great!

File: src/DrTests-RottenTests/DTRottenTestsPlugin.class.st [hunk 8]
```diff
@@ -37,12 +38,11 @@ DTRottenTestsPlugin >> run [
 			warning resume ] ]
 		on: RTFTestAnalysed
 		do: [ :notif | 
-			self ui
-				updateStatus: 'Analyzed test: {1}'
-				format: {notif compiledMethod printString}.
+			self announcer
+				announce: (DTStatusUpdate message: ('Analyzed test {1}.' format: {notif compiledMethod printString})).
```
Comment (guillep) [inline src/DrTests-RottenTests/DTRottenTestsPlugin.class.st:42]: Great!

File: src/DrTests-TestsRunner/DTTestsRunner.class.st [hunk 9]
```diff
@@ -58,10 +58,11 @@ DTTestsRunner >> runSuite: aTestSuite [
 DTTestsRunner >> runSuite: aTestSuite withResult: aResult [
 	aTestSuite
 		subscribe: TestAnnouncement
-		do: [ :testAnnouncement | self ui updateStatus: 'Running test {1}.' format: { testAnnouncement test asString } ].
-	[ aTestSuite run: aResult  ] ensure: [
-		aTestSuite unsubscribe:TestAnnouncement.
-	].
+		do: [ :testAnnouncement | 
+			self announcer
+				announce: (DTStatusUpdate message: ('Running test {1}.' format: {testAnnouncement test asString})) ].
```
Comment (guillep) [inline src/DrTests-TestsRunner/DTTestsRunner.class.st:63]: Great! :D

File: src/DrTests-Tests/DrTestsTestRunnerTest.class.st [hunk 10]
```diff
@@ -1,3 +1,6 @@
+"
+A DrTestsUITest is a test class for testing the behavior of DrTests-TestRunner
```
Comment (juliendelplanque) [inline src/DrTests-Tests/DrTestsTestRunnerTest.class.st:2]:
```suggestion
A DrTestsUITest is a test class for testing the behavior of DrTests-TestRunner.
```

File: src/DrTests-Tests/DrTestsTestRunnerUITest.class.st [hunk 11]
```diff
@@ -1,3 +1,6 @@
+"
+A DrTestsUITest is a test class for testing the behavior of DrTests-TestRunner UI
```
Comment (juliendelplanque) [inline src/DrTests-Tests/DrTestsTestRunnerUITest.class.st:2]:
```suggestion
A DrTestsUITest is a test class for testing the behavior of DrTests-TestRunner UI.
```

**Review by guillep — APPROVED** [review]
Looks very nice :)

File: src/DrTests/DTResultGroup.class.st [hunk 12]
```diff
@@ -1,3 +1,11 @@
+"
+I have the results of a DrTestPlugin analysis listed in groups: 
+-Errors
+-Failures
+-Skipped tests
+-PAssed test
```
Comment (juliendelplanque) [inline src/DrTests/DTResultGroup.class.st:6]:
```suggestion
-Passed test
```

File: src/DrTests/DTStatusUpdate.class.st [hunk 13]
```diff
@@ -0,0 +1,28 @@
+"
+Notify when executing an analisis.
```
Comment (juliendelplanque) [inline src/DrTests/DTStatusUpdate.class.st:2]:
```suggestion
Notify when executing an analysis.
```

File: src/DrTests/DrTestsPlugin.class.st [hunk 14]
```diff
@@ -1,8 +1,13 @@
+"
+I am a Plugin used to execute an analysis
+
+For run I need a DTPluginConfiguration that contains the items I will analyse. I define the way to execute the analysis and create a result (could be a TestResult).
```
Comment (juliendelplanque) [inline src/DrTests/DrTestsPlugin.class.st]:
```suggestion
To run, I need a DTPluginConfiguration that contains the items I will analyse. I define the way to execute the analysis and create a result (could be a TestResult). 
```

File: src/DrTests/DrTestsPlugin.class.st [hunk 15]
```diff
@@ -66,14 +81,12 @@ DrTestsPlugin >> resultButtonLabel [
 
 { #category : #api }
 DrTestsPlugin >> run [
```
Comment (juliendelplanque) [inline src/DrTests/DrTestsPlugin.class.st:83]: In fact, `#run` method can be totally removed it has been replaced by `#runForConfiguration:`.

File: src/DrTests/DrTestsUI.class.st [hunk 16]
```diff
@@ -1,3 +1,21 @@
+"
+I provide the ability to: 
+* select a plugin to create/run tests
+* select sets of items to analyze
+* obtain a detailed log of the results 
+
+UI Description 
+___________ 
+
+The droplist contains all the plugins avaible to start the analisis.
```
Comment (juliendelplanque) [inline src/DrTests/DrTestsUI.class.st]:
```suggestion
The droplist contains all the plugins available to start the analysis. 
```

File: src/DrTests/DrTestsUI.class.st [hunk 17]
```diff
@@ -1,3 +1,21 @@
...
+* select a plugin to create/run tests
+* select sets of items to analyze
+* obtain a detailed log of the results 
+
+UI Description 
+___________ 
+
+The droplist contains all the plugins avaible to start the analisis. 
+
+My left-most pane lists all of the categories that contain items (could subclasses of TestCase, executable comments, et c); Once items are selected, the items that can be analyzed appear in the pane to right.
```
Comment (juliendelplanque) [inline src/DrTests/DrTestsUI.class.st]:
```suggestion
My left-most pane lists all of the categories that contain items (could subclasses of TestCase, executable comments, etc.); Once items are selected, the items that can be analyzed appear in the pane to right.
```

File: src/DrTests/DrTestsUI.class.st [hunk 18]
```diff
@@ -1,3 +1,21 @@
...
+The right-most pane shows the results in diferent groups, depends the plugin's analisis.
```
Comment (juliendelplanque) [inline src/DrTests/DrTestsUI.class.st]:
```suggestion
The right-most pane shows the results in different groups, depends the plugin's analysis. 
```

**Review by juliendelplanque — CHANGES_REQUESTED** [review]
Will be integrated once review points have been fixed.

**Review by juliendelplanque — APPROVED** [review]

**Pull request merged**